### PR TITLE
Remove extra logging of ClientConnectorError

### LINKED ISSUE #257: Unnecessary logging of ClientConnectorError
While waiting for transaction confirmation, plugin reports errors with stacktraces when connection to the sequencer fails:
```
StarknetPluginError: Got ClientConnectorError while trying to access https://alpha-mainnet.starknet.io/feeder_gateway/get_transaction_status?transactionHash=0x33db7d1e35ab71f85d9ae1c0f9966c2ab986d73d2b7498616f8d78230aac2fc.
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/aiohttp/connector.py", line 1154, in _create_direct_connection
...
```
connection is later retired:
```
Retrying transaction status check...
```
Unnecessary stacktraces make logs difficult to read.

### PR DESCRIPTION
**Pull request by Nathan-SL**
## Usage related changes

-   Closes #257 
-   Removes error log coming from tx status checking in `iterativelyCheckStatus()`
-   Retries status checking a limited number of times (10)

## Development related changes

-   NA

## Checklist:

-   [x] Formatted the code
-   [x] No linter errors + tried to avoid introducing linter warnings
-   [x] Performed a self-review of the code
-   [x] Rebased to the last commit of the target branch (or merged it into my branch)
-   [x] Linked issues which this PR resolves
-   [x] All tests are passing (for external contributors who don't have access to the CI/CD pipeline)

### CODE REVIEW

**Review by FabijanC — CHANGES_REQUESTED** [review]

File: src/types.ts [hunk 1]
```diff
@@ -163,13 +163,13 @@ export async function iterativelyCheckStatus(
...
+    const statusObject = await checkStatus(txHash, starknetWrapper).catch(() => {
+        warn("Checking transaction status failed.");
         return undefined;
     });
 
     if (!statusObject) {
-        warn("Retrying transaction status check...");
+        warn("Retrying...");
         // eslint-disable-next-line prefer-rest-params
         setTimeout(iterativelyCheckStatus, CHECK_STATUS_RECOVER_TIMEOUT, ...arguments);
```
[the remark this answers is not in the document]
Comment (Nathan-SL) [reply src/types.ts]: https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/279/commits/48815a86f2fde5748316c9b47ac0f17c5e91182d

File: src/types.ts [hunk 2]
```diff
@@ -161,17 +161,31 @@ export async function iterativelyCheckStatus(
...
+        if (statusObject) {
+            break;
+        }
 
-    if (!statusObject) {
         warn("Retrying transaction status check...");
-        // eslint-disable-next-line prefer-rest-params
-        setTimeout(iterativelyCheckStatus, CHECK_STATUS_RECOVER_TIMEOUT, ...arguments);
+        await sleep(CHECK_STATUS_RECOVER_TIMEOUT);
+        count--;
```
Comment (FabijanC) [inline src/types.ts:183]: On the last iteration, we wait for `CHECK_STATUS_RECOVER_TIMEOUT`, only to not retry. The loop could be a `while true`, and in the end we could: decrement, check if count > 0 then either sleep or break out.

File: src/types.ts [hunk 3]
```diff
@@ -183,10 +197,8 @@ export async function iterativelyCheckStatus(
         );
     } else {
         // Make a recursive call, but with a delay.
-        // Local var `arguments` holds what was passed in the current call
-
-        // eslint-disable-next-line prefer-rest-params
-        setTimeout(iterativelyCheckStatus, CHECK_STATUS_TIMEOUT, ...arguments);
+        await sleep(CHECK_STATUS_TIMEOUT);
+        iterativelyCheckStatus(txHash, starknetWrapper, resolve, reject, retryCount);
```
Comment (FabijanC) [inline src/types.ts:201]: Why use recursion here? Can't all of this be done in a `while true` loop? Iterating in a loop should be computationally cheaper than recursing?

File: src/types.ts [hunk 4]
```diff
@@ -161,17 +161,31 @@ export async function iterativelyCheckStatus(
...
-    const statusObject = await checkStatus(txHash, starknetWrapper).catch((reason) => {
-        warn(reason);
-        return undefined;
-    });
+    let statusObject;
+    let errorReason;
+    let count = retryCount;
+
+    while (count > 0) {
+        statusObject = await checkStatus(txHash, starknetWrapper).catch((reason) => {
```
Comment (FabijanC) [inline src/types.ts:172]: A comment could be added before this line that this promise is rejected usually if the network is unavailable.

**Review by FabijanC — CHANGES_REQUESTED** [review]

File: src/types.ts (hunk 2 above)
Comment (Nathan-SL) [reply src/types.ts:194]: https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/279/commits/57b63a3440197fea0fc2600adf637b11376c2d4f

File: src/types.ts (hunk 3 above)
Comment (Nathan-SL) [reply src/types.ts]: https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/279/commits/57b63a3440197fea0fc2600adf637b11376c2d4f

File: src/types.ts (hunk 2 above)
Comment (FabijanC) [reply src/types.ts:194]: I think there was a misunderstanding, I didn't mean that the countdown loop should cover the whole function body, just that the recursion should be replaced with a loop. And the inner countdown loop should stay, no?

**Review by FabijanC — CHANGES_REQUESTED** [review]

File: src/types.ts [hunk 5]
```diff
@@ -149,45 +148,51 @@ async function checkStatus(hash: string, starknetWrapper: StarknetWrapper): Prom
 
 const ACCEPTABLE_STATUSES: TxStatus[] = ["PENDING", "ACCEPTED_ON_L2", "ACCEPTED_ON_L1"];
 export function isTxAccepted(statusObject: StatusObject): boolean {
-    return ACCEPTABLE_STATUSES.includes(statusObject.tx_status);
+    return ACCEPTABLE_STATUSES.includes(statusObject?.tx_status);
```
Comment (FabijanC) [inline src/types.ts:151]: Why did the question marks become necessary suddenly?

Comment (Nathan-SL) [reply src/types.ts]: Inside the `do while()` loop the two functions `isTxAccepted` & `isTxRejected` didn't return when `statusObject` was `undefined`, but removed them now. https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/279/commits/d4bf1b9755dba3abb800cf20914330786858b2ad

**Review by FabijanC — APPROVED** [review]

**Pull request merged**